app/actions/views/channel.js

```diff
@@ -155,6 +155,26 @@ export function loadPostsIfNecessary(channelId) {
     };
 }
 
+export function loadPostsIfNecessaryWithRetry(channelId) {
+    return (dispatch, getState) => {
+        const state = getState();
+        const {posts, postsInChannel} = state.entities.posts;
+
+        const postsIds = postsInChannel[channelId];
+
+        // Get the first page of posts if it appears we haven't gotten it yet, like the webapp
+        if (!postsIds || postsIds.length < ViewTypes.POST_VISIBILITY_CHUNK_SIZE) {
+            getPostsWithRetry(channelId)(dispatch, getState);
```

Comment:
With the retry methods we aren't waiting on a return really, just more so watching for the request to complete successfully, the retry attempts to be incremented, or the request to completely fail. We need to use the promise approach since we have to dispatch two actions, one for the request to start and the other for the retry method.